### ISSUE TITLE
nRemoteNG only support .Net6.0

### ISSUE THREAD

**Issue by MatthiasRecknagel** [mRemoteNG/mRemoteNG#2658]
## Current Behavior
nRemoteNG only support .Net6.0, but  .Net6.0 isn''t out of date, no updates available since 2024-11

## Your Environment

* Version used: Nightly and older
* Operating System and version:  Windows Server 16 x64

**Comment by simonai1254** [comment]
Hi Mathias. 

You phrased it somewhat unlucky, what you are trying to say is that .NET 6.0 reached EOL last month (12th November 2024 to be exact - see https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core). 

Therefore the project should be adapted to use the newer frameworks like .NET 8 or 9.

**Comment by Kvarkas** [comment]
where is a preparation happening to switch to .net8 due what release is a bit postnoted, as well expecting version shifting as well, stay tuned

**Comment by tecxx** [comment]
Hello,
i have upgraded the codebase to .net 8, here's a quick summary

**issue1** there are multiple locations in the code where a reference to "Windows.something" fails due to a namespace conflict. this is easily solved by prefixing with "App." -- example:
![image](https://github.com/user-attachments/assets/cad37caa-e1e9-42b9-8360-98d338ac2c00)


then the code compiles and mremoteng starts. unfortunately the treeview doesn't work. 

**issue2** this is caused by the 3rdparty "ObjectListView" package which is tageted to .net core 3.1 and doesn't seem to be maintained any longer. i was able to trace the issue down to VirtualListSize, funnily enough someone else already encountered and reported this here:
https://github.com/ennerperez/ObjectListView/issues/4
i took the sourcecode and added it to mremoteng, while updating the library to .net 8. 
![image](https://github.com/user-attachments/assets/9611563a-0c9a-4fb7-9283-32fd320a1fd7)

this fails at first because "MethodInvoker" is now ambiguous, but it can be fixed by adding the following to two files:
using MethodInvoker = System.Windows.Forms.MethodInvoker;

**issue3** finally, BinaryFormatter is deprecated and no longer usable in .net8, unless this is added to the project:

![image](https://github.com/user-attachments/assets/84aef510-8c16-49b8-b644-b01ea8651a3d)

as reported here: https://steven-giesel.com/blogPost/4271d529-5625-4b67-bd59-d121f2d8c8f6
this specific issue is a roadblock for migrating to .net9, so this needs to be solved at some point


the code changes are here for now -> [repo deleted]
i'll test the binary for a few days. shall i submit the changes to this repo later on with a PR?

**Comment by tecxx** [comment]
first test result, seems to work fine on hardware, on virtual machines i get an exception on startup here:
`Application: mRemoteNG.exe
CoreCLR Version: 8.0.1124.51707
.NET Version: 8.0.11
Description: The process was terminated due to an unhandled exception.
Exception Info: System.InvalidOperationException: Failed to retrieve disk identifier. Please ensure the disk information is accessible.
   at LocalSettingsDBManager.GetDiskIdentifier() in C:\SVN\mRemoteDev\mRemoteNG\Config\Settings\LocalSettingsManager.cs:line 394
`

![image](https://github.com/user-attachments/assets/1c90fbb4-b70f-494b-aae1-c61807f1c7b1)

seems like the VM does not report a serial number for it's disk, not sure why this was not in issue in the .net 6 based code... updated the code with a fix.
![image](https://github.com/user-attachments/assets/8bfa7c97-97c3-4653-b036-1ed23d2ab99e)

**Comment by Kvarkas** [comment]
what could be due a different method on .net8 so its dont match and dont receive what it expects, this is part of securing access to connections file what could be reworked. Will need to test on VM to see how

**Comment by tecxx** [comment]
on the vm it simply seems to return null instead of a serial number (see wbemtest-screenshot).  in .net 6 it probably returned empty string, so it didn't crash. i made a quickfix with a null check. works fine for now.

**Comment by Kvarkas** [comment]
if you wish to lead transition to .net 8 I may create a spare branch for such? also maybe worth to skip .net 8 and jump strait to .net 9?

**Comment by Kvarkas** [comment]
.net upgrade [branch](https://github.com/mRemoteNG/mRemoteNG/tree/v1.78.1-dev)

**Comment by tecxx** [comment]
i PR-ed my changes to the new branch, if i understood you right?
.net 9 requires a replacement for BinaryFormatter. would not focus on that now.

**Comment by txskibum** [comment]
@tecxx , I am not a developer or a programmer, but a mRemote user that is running into this issue.  I am using mRemote v1.77.3.1784-NB and .NET 8.0.11.  Seems mRemote is completely broken without the end of life .NET 6. I am not familiar enough with how GitHub works during development.  Do you have a new executable, DLLs, or any other files I could simply download and replace in the C:\Program Files\mRemote directory?  Any help, in laymen's terms, is greatly appreciated.

**Comment by tecxx** [comment]
@txskibum  until my changes get merged to the main repo, you can download the most recent sourcecode from my own fork -> [RepoDeleted - code is now in offical project] - compile it with visual studio 2022.

if you can't / don't want to do that, and trust a stranger on the internet, here is a download for the latest portable build (.net 8.0.12) - https://drive.google.com/file/d/1lTXjatriCwPLvIqItGhr_f-KnWlbHPKj/view?usp=sharing
please note that the portable build does not have an installer. just unpack the 7z file to a folder of your liking (e.g. desktop/mremote8).

**Comment by Kvarkas** [comment]
Yes, ok

**Comment by jsantoro339as** [comment]
Can anyone let me know when the .net 8 supported version will be available for public download? We are not able to use .net 6 due to our environment, so we are forced to uninstall .net 6.

**Comment by Kvarkas** [comment]
we have a new branch for .net 8 but its not yet production ready

**Comment by tecxx** [comment]
actually, i wanted to give feedback on this. we're using the .net8 version for a week internally and with customers. there were no issues reported. do you have a different experience?

**Comment by Kvarkas** [comment]
In terms of my project plans its dosnt have last improvements what I am still testing on .net6 (idea to finish to let somebody use it in case of they need access old machines, and then jump on new branch to add other fixes and improvements

**Comment by jsantoro339as** [comment]
@Kvarkas understood, so technically, 2 avail versions, one that supports .net 6 and one that supports .net 8.

Thanks for the update

**Comment by Kvarkas** [comment]
yes, .net 6 have to much limitation, but .net 8 dosnt suport windows 7 and older servers 2012

**Comment by Alessio71** [comment]
I managed to install 1.78.1.2856 (I cannot find it anymore), everything works fine, the only issue I'm facing is that I cannot enter a password for existing connections or new connections, the field password is grayed out and (visually) filled with stars.
Thank you.

**Comment by tecxx** [comment]
@Alessio71 
you're correct, thanks for reporting.
i didn't notice that because i don't store passwords locally. when i find time i try to find out what is causing this issue.

**Comment by Kvarkas** [comment]
> I managed to install 1.78.1.2856 (I cannot find it anymore), everything works fine, the only issue I'm facing is that I cannot enter a password for existing connections or new connections, the field password is grayed out and (visually) filled with stars. Thank you.

this is know issue, we implement secure sting to avoid potential passwords leakage to memory, however side effect is what listview then make field read only, and I dont find a solution so far for this.

**Comment by Kvarkas** [comment]
> [@Alessio71](https://github.com/Alessio71) you're correct, thanks for reporting. i didn't notice that because i don't store passwords locally. when i find time i try to find out what is causing this issue.

if you have time please check secure string implementation, what change listview field behavior

**Comment by stm69** [comment]
> yes, .net 6 have to much limitation, but .net 8 dosnt suport windows 7 and older servers 2012

Who cares? These Windows versions are out of support. Someone want to use mremoteng on such systems doesn't care about security and can use the old version with open CVE-2023-30367

Those who take care on security and use the maintained Windows version can not use mremoteng anymore because it has security issue to steal passwords from memory.

We have an IT team of about 50 peoples, until monthes ago we used mremoteng. To last year's end we had to find another solution because mremoteng looks like unmaintained, like dead, last official release six years old. Now we bought Royal TS for all admins and we are not that happy as we used to be with mremoteng.

**Comment by tecxx** [comment]
regarding securestring, and the non-editable password field: 
i won't fix this or work on this issue. the main reason is that it doesn't make sense anyway. the code is full of places where "securestring" is converted to or from a normal string, and even microsoft does not recommend to use this class, see:  https://learn.microsoft.com/en-us/dotnet/fundamentals/runtime-libraries/system-security-securestring

![Image](https://github.com/user-attachments/assets/85204db2-6a32-423d-ae01-543f3ddf56de)
 
i strongly believe that trying to develop a "secure" storage for remote connection information on the local machine is always going to fail. there is a reason enterprise password vaults exist, thats why i added two password vault connectors to mremote. (and even these have security limitations, see mimikatz and credssp issue when doing RDP)


TLDR; get rid of securestring and get a working release out.

**Comment by tecxx** [comment]
if anyone wants to build themselves, here's the .net8 compatible version with editable password field (securestring removed): https://github.com/tecxx/mRemoteNGtecxxDev

**Comment by Kvarkas** [comment]
As was mentioned early issue have deep roots, so it usable but for testing env only as without securestring passwords are open for memory dump, at this moment working to change how mR deal with credentials so problem will be solved
